docs/rules/no-implicit-coercion.md

```diff
@@ -33,10 +33,11 @@ This rule is aimed to flag shorter notations for the type conversion, then sugge
 
 This rule has three main options and one override option to allow some coercions as required.
 
-* `"boolean"` (`true` by default) - When this is `true`, this rule warns shorter type conversions for `boolean` type.
-* `"number"` (`true` by default) - When this is `true`, this rule warns shorter type conversions for `number` type.
-* `"string"` (`true` by default) - When this is `true`, this rule warns shorter type conversions for `string` type.
-* `"allow"` (`empty` by default) - Each entry in this array can be one of `~`, `!!`, `+` or `*` that are to be allowed.
+-   `"boolean"` (`true` by default) - When this is `true`, this rule warns shorter type conversions for `boolean` type.
+-   `"number"` (`true` by default) - When this is `true`, this rule warns shorter type conversions for `number` type.
+-   `"string"` (`true` by default) - When this is `true`, this rule warns shorter type conversions for `string` type.
+-   `"templateString"` (`false` by default) - When this is `true`, this rule warns shorter type conversions using template strings.
+-   `"allow"` (`empty` by default) - Each entry in this array can be one of `~`, `!!`, `+` or `*` that are to be allowed.
```

Comment:
This is a correct way to add this check in a minor version, but the options are getting confusing.

Disallowing `` `${foo}` `` when `"string"` is set to `true` and unless there's a corresponding entry in the `"allow"` array would be more in line with the existing schema. However, that would probably have to wait for the major version, and we also prefer to add new checks behind an option first and make it default later, so I'm not sure what's the best way to do this.

Thoughts?

It seems that the schema for this rule makes it difficult to add new checks like this in a minor version while keeping the options consistent.

Reply:
I agree. Backwards compatibility is the sole reason I added this as a new option instead of using the existing `string` option.

Reply:
Is this discussion blocking? Is any action required from me to get this merged?

Reply:
It seems that neither of the solutions is ideal, so let's add a new boolean option as proposed.

Can we name it something like `disallowTemplateShorthand` instead of `templateString`? The other three options `boolean`, `number`, and `string` refer to types, while this one refers to a notation.

Reply:
I renamed the option to `disallowTemplateString`.